File: languages/csharp/AppEncryption/Crypto/Exceptions/AppEncryptionException.cs
using System;

namespace GoDaddy.Asherah.Crypto.Exceptions
{
    public class AppEncryptionException : SystemException
    {
        public AppEncryptionException(string message)
            : base(message)
        {
        }

        public AppEncryptionException(string message, Exception cause)
            : base(message, cause)
        {
        }
    }
}

File: java/app-encryption/src/main/java/com/godaddy/asherah/appencryption/exceptions/AppEncryptionException.java
package com.godaddy.asherah.appencryption.exceptions;

public class AppEncryptionException extends RuntimeException {

  /**
   * Creates a new {@code AppEncryptionException}. This signals that a
   * {@link com.godaddy.asherah.appencryption.envelope.EnvelopeEncryption} exception has occurred.
   *
   * @param message The detailed exception message.
   */
  public AppEncryptionException(final String message) {
    super(message);
  }

  /**
   * Creates a new {@code AppEncryptionException}. This signals that a
   * {@link com.godaddy.asherah.appencryption.envelope.EnvelopeEncryption} exception has occurred.
   *
   * @param message The detailed exception message.
   * @param cause The actual {@link java.lang.Exception} raised.
   */
  public AppEncryptionException(final String message, final Exception cause) {
    super(message, cause);
  }
}
